src/library_pthread.js

```diff
@@ -1069,6 +1084,80 @@ var LibraryPThread = {
 #endif
   },
 
+#if MAIN_MODULE
+  _emscripten_thread_exit_joinable: function(thread) {
+    // Called when a thread exits and is joinable.  This puts the thread
+    // into zombie state where it can't run anymore work but cannot yet
+    // be cleaned up.
+    if (!ENVIRONMENT_IS_PTHREAD) markAsZombie(thread);
+    else postMessage({ 'cmd': 'markAsZombie', 'thread': thread });
```

Comment:
Could this use `proxy: async` rather than manually postMessage?

Reply:
This is following the internal style for all the other similar thread functions.  It does looks like we should be able ot use the `__proxy` system, so I will look into doing that as a separate PR that can land ahead of this one.